adafruit_led_animation/animation/sparkle.py

```diff
@@ -57,18 +57,22 @@ class Sparkle(Animation):
     :param float speed: Animation speed in seconds, e.g. ``0.1``.
     :param color: Animation color in ``(r, g, b)`` tuple, or ``0x000000`` hex format.
     :param num_sparkles: Number of sparkles to generate per animation cycle.
+    :param mask: array to limit sparkles within range of the mask
     """
 
     # pylint: disable=too-many-arguments
-    def __init__(self, pixel_object, speed, color, num_sparkles=1, name=None):
+    def __init__(self, pixel_object, speed, color, num_sparkles=1, name=None, mask=[]):
         if len(pixel_object) < 2:
             raise ValueError("Sparkle needs at least 2 pixels")
+        if len(mask) >= len(pixel_object):
+            raise ValueError("Sparkle mask should be smaller than number pixel array")
         self._half_color = color
         self._dim_color = color
         self._sparkle_color = color
         self._num_sparkles = num_sparkles
         self._num_pixels = len(pixel_object)
         self._pixels = []
+        self._mask = mask
```

Comment:
here we can check for `None` and set it up accordingly like this:
```python
if mask:
    self._mask = mask
else:
    self._mask = []

if len(self._mask) >= len(pixel_object):
    raise ValueError("Sparkle mask should be smaller than number pixel array")
```

Reply:
makes sense, will fix. thanks...